code/script.js

```diff
@@ -269,57 +286,90 @@ const filterCharacters = (keep) => {
+    } else {
+      alert(
+        `No, wrong! ${value} is not correct, toss all the ${value}haired people!`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+  } else if (category === 'eyes') {
+    if (keep) {
+    alert(
+      `Yes, our secret person does have ${value} eyes. So long to all the other ones!`
+      )
+    charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(
+        `Wrong! Our secret person does not have ${value} eyes. Say godbye to all people with ${value} eyes!`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  generateBoard()
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+const guess = (personToConfirm) => { 
+  guessedPerson = confirm(`Are you sure you want to guess ${personToConfirm}?`)
+ 
+  if (guessedPerson) {
+     checkMyGuess(personToConfirm)
+  } else {
+     alert('Ok, go back at it!')
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+    const {name} = secret
+    board.style.display = 'none'
+    winOrLose.style.display = 'flex'
+    
+    if (personToCheck === name) {
+     winOrLoseText.innerText = `Amazeballs, that's correct! Do it again!`
+  } else { (guess !== secret)
+     winOrLoseText.innerText = `Noooo, that's wrong! Do it again!`
+  }
 }
 
 // Invokes the start function when website is loaded
 start()
 
 // All the event listeners
 restartButton.addEventListener('click', start)
+findOutButton.addEventListener('click', checkQuestion)
+questions.addEventListener('change', selectQuestion)
+playAgainBtn.addEventListener('click', () => {
+  winOrLose.style.display = "none";
+  board.style.display = "flex";
```

Comment:
Oh this was a nice detail, I didn't think of putting the changes in display with the eventLinstener. SMART!!!